### ISSUE TITLE
authFunction setting is missing in DAVClient constructor

### ISSUE THREAD

**Issue by general-ice** [natelindev/tsdav#191]
Hey! First of all thanks for the great tool.

I guess there is missing authFunction assignment in DAVClient constructor. It passed in a params but does's save anywhere. Look.

_src/client.ts_
```
export class DAVClient {
  ...
  constructor(params: {
    serverUrl: string;
    credentials: DAVCredentials;
    authMethod?: 'Basic' | 'Oauth' | 'Digest' | 'Custom';
    authFunction?: (credentials: DAVCredentials) => Promise<Record<string, string>>; // Not saving
    defaultAccountType?: DAVAccount['accountType'] | undefined;
  }) {
    this.serverUrl = params.serverUrl;
    this.credentials = params.credentials;
    this.authMethod = params.authMethod ?? 'Basic';
    this.accountType = params.defaultAccountType ?? 'caldav';
  }
...
}
```

At the same time other methods try to use that property, but it constantly equals undefined.

**Comment by natelindev** [comment]
fixed in v2.0.10

**Comment by darshan-mundra-08** [comment]
I am still facing this error for my project
